### DISCUSSION TITLE
Getting transcript for longer videos

### DISCUSSION THREAD

**Discussion by agnirudrasikdar** [jdepoix/youtube-transcript-api#265] [General]
Firstly, I want to thank you for the amazing library! This is great work! 
I was testing out with the library and found that for longer videos, the transcript kind of cuts off after ~40mins . Is this a limitation with the current library? What will the solution to get the complete transcript for long videos.  
Many thanks prior

**Comment by jdepoix** [comment]
Hi @agnirudrasikdar!
I don't have much experience with videos that long, could you provide a few examples?

**Reply by agnirudrasikdar** [reply]
@jdepoix 
Example:
a) https://www.youtube.com/watch?v=TDOMekBPR4U
b) https://www.youtube.com/watch?v=rH1Y66IwKvc&rco=1
c) https://www.youtube.com/watch?v=tKdI3kvHZJY
d) https://www.youtube.com/watch?v=hInAmmQSfI4

(and similar length videos) 

It seems beyond the 40th minute mark, the transcript just cuts off abruptly.

**Reply by ByteToHex** [reply]
> @jdepoix Example: a) https://www.youtube.com/watch?v=TDOMekBPR4U b) https://www.youtube.com/watch?v=rH1Y66IwKvc&rco=1 c) https://www.youtube.com/watch?v=tKdI3kvHZJY d) https://www.youtube.com/watch?v=hInAmmQSfI4
> 
> (and similar length videos)
> 
> It seems beyond the 40th minute mark, the transcript just cuts off abruptly.

The issue of the 40min cutoff does not seem to exist anymore? I just tried it and I was able to programmatically retrieve everything up until the last transcript snippet (1:43:06 in for [[TDOMekBPR4U](https://www.youtube.com/watch?v=TDOMekBPR4U)], which is a)

However there seems to be an issue with retrieving [rH1Y66IwKvc](https://www.youtube.com/watch?v=rH1Y66IwKvc&rco=1) as I get the following error

> ::Video ID: rH1Y66IwKvc
> Traceback (most recent call last):
>   File "d:\WS\(GH)DEV-Ref\WS-Python\Tasks\Manage-Text\Retrieve_Youtube_Transcript\Retrieve_Yt_Transcript.py", line 69, in <module>
>     transcript_data = fetch_transcript(video_id, languages)
>   File "d:\WS\(GH)DEV-Ref\WS-Python\Tasks\Manage-Text\Retrieve_Youtube_Transcript\Retrieve_Yt_Transcript.py", line 41, in fetch_transcript
>     transcript_list = ytt_api.list(video_id)
>   File "C:\ProgramData\anaconda3\envs\env\Lib\site-packages\youtube_transcript_api\_api.py", line 127, in list
>     return self._fetcher.fetch(video_id)
>            ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
>   File "C:\ProgramData\anaconda3\envs\env\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 356, in fetch
>     self._fetch_captions_json(video_id),
>     ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
>   File "C:\ProgramData\anaconda3\envs\env\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 364, in _fetch_captions_json
>     return self._extract_captions_json(innertube_data, video_id)
>            ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\ProgramData\anaconda3\envs\env\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 385, in _extract_captions_json
>     self._assert_playability(innertube_data.get("playabilityStatus"), video_id)
>     ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\ProgramData\anaconda3\envs\env\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 420, in _assert_playability
>     raise VideoUnplayable(
>         video_id, reason, [run.get("text", "") for run in subreasons]
>     )
> youtube_transcript_api._errors.VideoUnplayable:
> Could not retrieve a transcript for the video https://www.youtube.com/watch?v=rH1Y66IwKvc! This is most likely caused by:
> 
> The video is unplayable for the following reason: The following content may contain suicide or self-harm topics.
> 
> If you are sure that the described cause is not responsible for this error and that a transcript should be retrievable, please create an issue at https://github.com/jdepoix/youtube-transcript-api/issues. Please add which version of youtube_transcript_api you are using and provide the information needed to replicate the error. Also make sure that there are no open issues which already describe your problem!

**Reply by jdepoix** [reply]
@ByteToHex the error you have reported is unrelated to this discussion. Please open a new issue for that!
